fix: build-script for ubuntu 16.04 and fedora 30

amitsadaphule opened the pull request:
#### Description of Change

Fixes the build for Fedora 30 and Ubuntu 16.04 (https://github.com/leo-lb/electron/issues/1).

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes

Notes: `no-notes`

amitsadaphule committed:
fix: build-script for ubuntu 16.04 and fedora 30

llebout commented on ppc64le_build.bash:
```diff
@@ -1,22 +1,50 @@
+OS=$NAME
+OSVER=$VERSION_ID
+
+if [[ $OS == "Ubuntu" ]] && [[ $OSVER == "16.04" ]]
```
While it's good you choose to add support for Ubuntu Xenial, can't we keep Ubuntu Bionic even if it couldnt build Chromium lately?

amitsadaphule replied:
Thanks for the review @leo-lb! The reason why I didn't add bionic was that it had build issues and I couldn't really get it to build/test. But as you said, I'll just let bionic build too to enter there.

llebout replied:
"enter here" you mean in the Xenial code path? Dependencies may be different. - Just add a condition that retains the old code that installed dependencies, is there any problem with that?

amitsadaphule replied:
OK, it seems that I had misinterpreted your comment. I'll add separate condition for bionic. Also, the issue is not with the dependencies, I've used mostly the same dependencies for xenial (except for python3.7 and ninja). @NishikantThorat had tried the bionic build and had faced issues in the build, so we didn't really pursue it further.

amitsadaphule replied:
I was able to build on bionic too.

llebout commented on ppc64le_build.bash:
```diff
@@ -1,22 +1,50 @@
 wget "https://nodejs.org/dist/$VERSION/node-$VERSION-$DISTRO.tar.xz"
 tar -xJvf node-$VERSION-$DISTRO.tar.xz
 PATH=$(pwd)/node-$VERSION-$DISTRO/bin:$PATH
+ln -sf $(pwd)/node-$VERSION-$DISTRO/bin/node /usr/bin/node
```
You should not create a link in a system directory, manipulating the PATH variable and patching build scripts if they reference absolute paths should be sufficient.

We do not want the build script to require being run as root

amitsadaphule replied:
OK, I'll try to fix the reference to absolute path '/usr/bin/node' in the build. Same thing will apply for the python3.7 ubuntu link too. Will work on that one too.

llebout replied:
Didnt see the Python one, yes please do!

amitsadaphule replied:
Completed the necessary changes. Awaiting approval for contributing to https://github.com/leo-lb/chromium_power (specifically to https://github.com/leo-lb/chromium_power/blob/master/third_party/0001-DONTMERGE-third_party-node-Use-system-nodejs-binary.patch) w.r.t node path.

amitsadaphule replied:
Please review https://github.com/leo-lb/chromium_power/pull/3 for the node path fix.

llebout reviewed (changes requested):
Few comments, after that it's all good!

amitsadaphule committed:
Incorporate review comments

- change build-script to run as non-root user
- remove soft-link creation to node and python3.7

amitsadaphule commented:
The PR https://github.com/leo-lb/chromium_power/pull/3 along with the new commit a66b40b3afcaff3f261300c66759e43f41216c7d incorporates all the review comments.

leo-lb commented:
@amitsadaphule Merged!
